### ISSUE TITLE
was loaded over HTTPS, but requested an insecure frame

### ISSUE THREAD

**Issue by xanthos84** [airsonic/airsonic#1562]
## Problem description

I run Airsonic behind a caddy reverse proxy, and when I load the page, i get an 

Mixed Content: The page at 'https://URL/index' was loaded over HTTPS, but requested an insecure frame 'http://URL/home.view'. This request has been blocked; the content must be served over HTTPS.

error

### Steps to reproduce

setup caddy with simple config:

URL {
  proxy / IP-OF-AIRSONIC-HOST:PORT {
  websocket 
  transparent }
}

and call the URL

## System information

newest airsonic via docker image (linuxserver/airsonic)
with no more special configs on top
running on ubuntu 18.04 host

**Comment by fxthomas** [comment]
Have you looked at the [proxy documentation](https://airsonic.github.io/docs/proxy/prerequisites/) (although Caddy is missing), particularly the `X-Forwarded-Proto` header?

**Comment by xanthos84** [comment]
yes, with the "transparent" option as stated above all the parameters are already set,
but also on top tried to set them manually, and it changes nothing in the behaviour of airsonic delivering an http link.

**Comment by fxthomas** [comment]
I'm afraid I won't be of much help then, I have never tried Caddy...
If you do figure it out let us know so that we can add instructions in the documentation!

**Comment by jzinkweg** [comment]
With the latest release (based on Spring Boot 2.2) I had to add `-Dserver.forward-headers-strategy=framework` to the startup parameters.
(and in my case I had to add X-Forwarded-Proto/X-Forwarded-SSL to the headers sent by Apache)

**Comment by benschumacher** [comment]
I'm using Caddy (currently v0.11.3, w/ this [Docker image](https://quay.io/joshix/caddy)) with Airsonic. My `Caddyfile` has this configuration:

```
(shared) {
    errors /log/errors.log {
        rotate_size 100 # Rotate a log when it reaches 100 MB
        rotate_age  14  # Keep rotated log files for 14 days
        rotate_keep 10  # Keep at most 10 rotated log files
        rotate_compress # Compress rotated log files in gzip format
    }
    log stdout
    tls {
	max_certs 2
    }
}

airsonic.(private) {
    import shared

    proxy / airsonic:4040 {
        transparent
    }
}
```

And I'm running Airsonic using the pre-built Docker with these JAVA_OPTS in command-line:

```
--env="JAVA_OPTS=-Dserver.use-forward-headers=true -Xmx256m"
```

**Comment by fxthomas** [comment]
The [Caddy docs](https://airsonic.github.io/docs/proxy/caddy/) actually exist now, and they specify clearly that you have to use `server.use-forward-headers` (actually this is true for all proxy servers). Let me know if you feel something is missing, and whether you fixed your initial issue?

The Spring 2.2 update changed the name of a few properties, if you're running on `master` you might need to use `server.servlet.contextPath` instead of `server.context-path`. Other than that it's working fine on my instance with other properties unchanged.

@jzinkweg Weird, I've never had to add `forward-headers-strategy`. It doesn't work for you without that?

**Comment by jzinkweg** [comment]
I was impacted by the context path change and while checking the docs I found this:

> The server.use-forward-headers property has been deprecated in favor of server.forward-headers-strategy; previously server.use-forward-headers=true would use the web server native support. You can now achieve the same with server.forward-headers-strategy=native. Because each server has a specific behavior, we now offer an alternative which relies on Spring’s ForwardedHeaderFilter: server.forward-headers-strategy=framework. Developers should use the option that fits best their use case.

...so I made that change as well.
See: https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.2-Release-Notes#deprecations-in-spring-boot-22

**Comment by fxthomas** [comment]
Nice, thanks for looking into it! I added a todo so that we remember to add it in the release notes when 11.0 is out.
